### ISSUE TITLE
Assumes tape is small

### ISSUE THREAD

**Issue by rdebath** [kmyk/brainfuck-interpreter-analyzer#1]
This program goes wrong if the cell size is 32bit, 64bits, 128bit, floating point or unbounded (and probably others.)

It also can't cope with an interpreter that has a 1000000000 cells.

**Comment by kmyk** [comment]
Yes, this program cannot assume the correct cell size when it is large.
I have known this, but the output was much confusing. Fixed it.

About floating point cell, I didn't know this. And I implemented a checker for it.

Thank you :smile:

**Comment by rdebath** [comment]
You might try something like [this test](https://github.com/rdebath/Brainfuck/blob/master/testing/Cellsize.b), it detects large cells that are powers of two by using the optimisation that an interpreter can do. It always gives small results, if it gives a large result (32bit+) you know the interpreter is optimising too.
